physical/s3/s3_test.go

```diff
@@ -29,6 +31,10 @@ func DoS3BackendTest(t *testing.T, kmsKeyId string) {
 		t.Skip()
 	}
 
+	if !hasAWSCredentials() {
```

Comment:
I'll approve this, but note that I've been meaning for a while to adapt these tests to use a Minio docker container as a reasonable facsimile of S3, as we do for autosnapshots.  Which doesn't mean we shouldn't also do real tests with S3, but they're not mutually exclusive.